src/tasks/buildArtifact/BuildArtifactTaskProvider.ts

```diff
@@ -134,44 +141,52 @@ export class BuildArtifactTaskProvider implements TaskProvider {
+                const defaultDefinition: IExportJarTaskDefinition = {
+                    type: BuildArtifactTaskProvider.exportJarType,
+                    mainClass: (mainClasses.length === 1) ? mainClasses[0].name : undefined,
+                    targetPath: Settings.getExportJarTargetPath(),
+                    elements: elementList,
+                };
+                const defaultTask: Task = new Task(defaultDefinition, folder, folder.name, BuildArtifactTaskProvider.exportJarType,
+                    new CustomExecution(async (resolvedDefinition: IExportJarTaskDefinition): Promise<Pseudoterminal> => {
+                        const stepMetadata: IStepMetadata = {
+                            entry: undefined,
+                            taskLabel: resolvedDefinition.label || folder.name,
+                            workspaceFolder: folder,
+                            projectList: projectList,
+                            steps: [],
+                            elements: [],
+                            classpaths: [],
+                        };
+                        return new ExportJarTaskTerminal(resolvedDefinition, stepMetadata);
+                    }), undefined);
+                defaultTask.presentationOptions.reveal = TaskRevealKind.Never;
+                this.tasks.push(defaultTask);
+            } catch (e) {
+                continue;
             }
-            const mainClasses: IMainClassInfo[] = await Jdtls.getMainClasses(folder.uri.toString());
-            const defaultDefinition: IExportJarTaskDefinition = {
-                type: BuildArtifactTaskProvider.exportJarType,
-                mainClass: (mainClasses.length === 1) ? mainClasses[0].name : undefined,
-                targetPath: Settings.getExportJarTargetPath(),
-                elements: elementList,
-            };
-            const defaultTask: Task = new Task(defaultDefinition, folder, folder.name, BuildArtifactTaskProvider.exportJarType,
-                new CustomExecution(async (resolvedDefinition: IExportJarTaskDefinition): Promise<Pseudoterminal> => {
-                    const stepMetadata: IStepMetadata = {
-                        entry: undefined,
-                        taskLabel: resolvedDefinition.label || folder.name,
-                        workspaceFolder: folder,
-                        projectList: await Jdtls.getProjects(folder.uri.toString()),
-                        steps: [],
-                        elements: [],
-                        classpaths: [],
-                    };
-                    return new ExportJarTaskTerminal(resolvedDefinition, stepMetadata);
-                }), undefined);
-            defaultTask.presentationOptions.reveal = TaskRevealKind.Never;
-            this.tasks.push(defaultTask);
         }
         return this.tasks;
     }
+
+    private static getLanguageServerStatusTimeout(): Promise<boolean> {
```

Comment:
nit: just calling it `languageServerStatusTimeout()` should be fine. Since we do not get some real data here